src/ethereum/state_oracle/__init__.py

```diff
@@ -0,0 +1,47 @@
+"""
+State Oracle Interface for Ethereum Execution Specs.
+
+Provides an abstract interface for state access that can be implemented
+by different state management strategies.
+"""
+
+from typing import Optional
+
+from .interface import MerkleOracle
+from .memory_oracle import MemoryMerkleOracle
+
+_state_oracle: Optional[MerkleOracle] = None
+
+
+def set_state_oracle(oracle: MerkleOracle) -> Optional[MerkleOracle]:
+    """
+    Set the global state oracle.
+
+    Returns the previous oracle if any.
+    """
+    global _state_oracle
+    old = _state_oracle
+    _state_oracle = oracle
+    return old
+
+
+def get_state_oracle() -> MerkleOracle:
```

Comment:
You're still returning an object here. We want this to be as flat as possible. For each function on the `MerkleOracle`, create a function in this file exposing it.